fineract-investor/src/main/java/org/apache/fineract/investor/service/ExternalAssetOwnersReadServiceImpl.java

```diff
@@ -120,6 +120,13 @@ public ExternalTransferData retrieveFirstTransferByExternalId(ExternalId externa
                 .orElseThrow(() -> new ExternalAssetOwnerTransferNotFoundException(externalTransferId));
     }
 
+    @Override
+    public Long retrieveFirstTransferIdByExternalId(ExternalId externalTransferId) {
+        return externalAssetOwnerTransferRepository.findFirstByExternalIdOrderByIdAsc(externalTransferId)
```

Comment:
Would it make sense to find the latest? i dont really see the reason why to look for the first one... :/

Reply:
I think it's irrelevant because transferExternal id will identify only one transfer (or two if already cancelled), but Ofc we can get the latest...